rm/rm-server/src/main/java/org/ow2/proactive/resourcemanager/nodesource/infrastructure/SSHInfrastructureV2.java

```diff
@@ -51,10 +51,12 @@
 import org.ow2.proactive.resourcemanager.core.properties.PAResourceManagerProperties;
 import org.ow2.proactive.resourcemanager.exception.RMException;
 import org.ow2.proactive.resourcemanager.nodesource.common.Configurable;
-import org.ow2.proactive.resourcemanager.utils.CommandLineBuilder;
+import org.ow2.proactive.resourcemanager.nodesource.utils.InitScriptGenerator;
```

Comment:
instead of completely replacing CommandLineBuilder usage in the node source, why not add extra feature to it ?

something like:
`clb.useStartupScript(String scriptContent)`

then `clb.buildCommandLine()` will use the script to generate the command line instead of building the command directly.

Also: 
 * the node source should contain a nodeJarUrl parameter
 * some properties such as -Dproactive.communication.protocol=%protocol% should not be set automatically

Reply:
The main Idea here is that we are imitating the work done in the new node source in nodesource-addons, CommandLineBuilder mainly creates the command to start the node agent which was replaced with NodeSource.properties, a configurable string that will be filled by simple calls. 
CommandLineBuilder is specific for a single type of call (start a proactive node agent), and for example if we want to use node.jar we have to recreate the whole class. 
In contrast, InitScriptGenerator is more elegant, the startup script will be imported according to the mode of deployment you want and then the variable will simply be passed to replace the properties. So for the next task completely replacing the CommandLineBuilder is necessary. 
PLEASE NOTE:  the branch is not ready to be merged, I pushed it so Isabelle can take a look and give me feedback, some changes are still necessary.

Reply:
"the node source should contain a nodeJarUrl parameter" Yes this is left for the next step.

Reply:
ok for the branch not ready to be merged, but you need to go in the right direction directly.

I believe the right direction is to use CommandLineBuilder and add inside the necessary extra features.

In that case the extra features is to build a command from a properties file with pattern replacement.